allennlp/modules/stacked_elmo_lstm.py

```diff
@@ -0,0 +1,168 @@
+"""
+A stacked bidirectional LSTM with skip connections between layers.
+"""
+
+from typing import Optional, Tuple, List
+import torch
+from torch.nn.utils.rnn import PackedSequence, pad_packed_sequence
+from allennlp.modules.lstm_cell_with_projection import LSTMCellWithProjection
+from allennlp.common.checks import ConfigurationError
+
+
+class ElmoLstm(torch.nn.Module):
```

Comment:
Nit, but can you change this name to `ELMoLSTM`?  To be consistent with everything else.

Reply:
It's easier to read as it is, and fits with the rest of the codebase, so i'd rather not. Do you mean it's more consistent with your TF code?

Reply:
Consistent with the paper and everywhere else where we write `ELMo`.  It's an acronym.  If we applied your reasoning to all of the class names it'd also be `Lstm` right?

Reply:
Which I see you have done `Lstm` -- but pytorch itself uses `LSTM` as does just about every other library I've seen.

Reply:
The recommendation comes from here: https://google.github.io/styleguide/javaguide.html#s5.3-camel-case.  I think something like `XmlHttpRequest` is way easier to read than `XMLHTTPRequest` - in the second you have to do word segmentation as you read the identifier, while in the first you don't.

Reply:
Sure, but: (1) that's a Java style guide (2) allennlp doesn't follow any established python style guide that I can tell and (3) every other deep learning library uses `LSTM`.

Reply:
https://stackoverflow.com/questions/15526107/acronyms-in-camelcase

https://github.com/tensorflow/tensorflow/blob/18003982ff9c809ab8e9b76dd4c9b9ebc795f4b8/tensorflow/contrib/lite/models/speech_speakerid_model_test.cc#L38-L43
https://github.com/tensorflow/tensorflow/blob/5e6b16a9261d95eee321a67e38ea0a83357f3bf8/tensorflow/contrib/ndlstm/python/lstm2d_test.py#L64
https://github.com/tensorflow/tensorflow/blob/14f564990e7548b4a5c41da0fc533ec5a7a25abe/tensorflow/contrib/rnn/python/kernel_tests/rnn_cell_test.py#L225
https://github.com/tensorflow/tensorflow/blob/b10f50ff15944badb7262a207f6628dfa52d6a9d/tensorflow/contrib/ndlstm/python/misc_test.py#L36
https://github.com/tensorflow/tensorflow/blob/d4ad9c73969c45d1a224ebfc43eb645b9860216b/tensorflow/contrib/lite/kernels/internal/optimized/optimized_ops.h#L1914

Which is easier for you to read: `CRFLSTMModel` or `CrfLstmModel`?  You're right that it doesn't matter very much if you have an identifier that's just an acronym, like `LSTM`.  It matters a whole lot more when the acronym is just part of a longer identifier, like `ELMoLSTM`.  Where do the acronyms start and stop?  How many are there?  It unambiguously hurts readability to have to parse that yourself.  Whether other people follow best practices, or even their own style guides (Google's naming conventions on camel case are not unique to java), shouldn't dictate how we write our code.

For (2), we generally follow Google's style guide: https://google.github.io/styleguide/pyguide.html#Indentation.

Reply:
We clearly have different views on this, and it's your library, so I'll stop after this comment. 
 I think the overall goal should be to make the library as easy to use and contribute to as possible.  Using non-standard python things or inconsistency with the rest of the deep learning world hinders both goals.

In tensorflow, LSTM is always used in the public facing APIs even as part of longer strings (https://www.tensorflow.org/api_docs/python/tf/contrib/rnn).

At lower levels, they aren't consistent -- e.g. one of those test files mixes both `LSTM` and `Lstm`  (https://github.com/tensorflow/tensorflow/blob/14f564990e7548b4a5c41da0fc533ec5a7a25abe/tensorflow/contrib/rnn/python/kernel_tests/rnn_cell_test.py#L173).

Python adopts the full upper case for acronyms in the standard library, e.g.

https://docs.python.org/3/library/urllib.request.html#urllib.request.HTTPDefaultErrorHandler
https://docs.python.org/3/library/xmlrpc.server.html#xmlrpc.server.SimpleXMLRPCServer